extensions/data-plane/data-plane-http/src/main/java/org/eclipse/edc/connector/dataplane/http/params/decorators/BaseSourceHttpParamsDecorator.java

```diff
@@ -53,7 +53,7 @@ public HttpRequestParams.Builder decorate(DataFlowRequest request, HttpDataAddre
     }
 
     private @NotNull String extractMethod(HttpDataAddress address, DataFlowRequest request) {
-        if (Boolean.parseBoolean(address.getProxyMethod())) {
+        if (Boolean.parseBoolean(address.getProxyMethod()) && "HttpProxy".equals(request.getDestinationDataAddress().getType())) {
```

Comment:
I don't expect that this will solve the issue regarding the `HttpPush`. This will disable the feature of proxying http methods for the `HttpPush`, as now just "GET" requests will be sent to the providing backend.

Our expectation would be that http-methods also could be proxied for the `HttpPush`

Reply:
simply, for http push this condition won't ever met, and it will pass through line 60, so the method will be taken from the `DataAddress` (as it already does) if it's defined, otherwise will use `GET`.

To verify that, just try to remove the `proxyMethod` property and set `method` instead on the source data address.

Reply:
> simply, for http push this condition won't ever met, and it will pass through line 60, so the method will be taken from the DataAddress (as it already does) if it's defined, otherwise it will use the default GET.

Ohh yes, I have overlooked this: The `HttpPush` will use the `method` from the `DataAddress`.

> The proxy was made to work with pull, not for push

This would implicate that certain `HttpData` assets can just be used for `HttpProxy` transfer-types and not for `HttpData` transfer-types. Would't it make sense than to use `HttpProxy` type also for the providing asset?

Reply:
theoretically yes, I'm not sure which would be the applications for that: the consumer should be the one who chose how the call to its service will be made.

Reply:
The applications would be the same as for the `http-pull` I think. 

So for the `http-pull` the `consumer` is able to select the preferred http-method, which is used to access the `provider` system. This is enabled by using the `prxoyMethod=true` property in the asset of the provider.

For the `http-push` the transfer is also started by the `consumer`. Like for the `http-pull` the `consumer` should be able to select the prefered http-method, which is used to access the `provider` system.

Reply:
in any case, also without the `proxyMethod` flag to true, the consumer can choose the method, that by default is `GET`. Really, all the `proxyXXX` stuff is there for the `proxy` use case ("nomen omen").

Reply:
Hey @ndr-brt,

I'm sry for coming back on this so lately, but this change is really important to us and I may not have understood this correctly.

After giving this a second look, I am also convinced, that this change won't affect passing methods for the `HttpPush` from the consumer.

My problem was, that I though the `HttpDataAddress address` parameter is always populated by the `SourceDataAddress`:
https://github.com/eclipse-edc/Connector/blob/cf629c963c51f74157eef4d03f10dbf476c6e625/extensions/data-plane/data-plane-http/src/main/java/org/eclipse/edc/connector/dataplane/http/params/decorators/BaseSourceHttpParamsDecorator.java#L55-L61

but it turns out it also gets called for the `DestinationDataAddress`:
https://github.com/eclipse-edc/Connector/blob/cf629c963c51f74157eef4d03f10dbf476c6e625/extensions/data-plane/data-plane-http/src/main/java/org/eclipse/edc/connector/dataplane/http/params/HttpRequestParamsProviderImpl.java#L54-L68

So indeed the `extractMethod` returns the method specified by the `DestinationDataAddress`.

Did I understood this correctly?

Thanks for fixing this!